Sources/Datadog/DatadogCore/Context/LaunchTimeReader.swift

```diff
@@ -0,0 +1,41 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+#if SPM_BUILD
+import _Datadog_Private
+#endif
+
+internal struct LaunchTimeReader: ContextValueReader {
+    let initialValue: LaunchTime
+
+    /// Lock object to sync launch time read.
+    let lock: Any
+
+    init() {
+        let lock = NSObject()
+        initialValue = __dd_private_objc_sync_LaunchTime(lock)
+        self.lock = lock
+    }
```

Comment:
Shouldn't we define single, static `lock`?
```swift
internal struct LaunchTimeReader: ContextValueReader {
   private static lock = NSObject()

   // ...
}
```
Otherwise, we will be creating distinct `lock` object for each "instance" of `LaunchTimeReader`. This will become an issue when multiple instances of the SDK run in a single process - the critical section behind this `lock` will be accessed by multiple readers at once.

Reply:
We already use a static lock in [ObjcAppLaunchHandler](https://github.com/DataDog/dd-sdk-ios/blob/develop/Sources/_Datadog_Private/ObjcAppLaunchHandler.m#L89L91), so we are safe in case of multiple readers. This instance lock is coming from the previous implementation of [LaunchTimeProvider](https://github.com/DataDog/dd-sdk-ios/blob/develop/Sources/Datadog/Core/System/LaunchTimeProvider.swift#L28L36), which says:
```
// Even if __dd_private_AppLaunchTime() is using a lock behind the
// scenes, TSAN will report a data race if there are no
// synchronizations at this level.
```
I've tried removing this lock and I didn't see any TSAN error.. so I'm not sure where it's coming from, do you know more?
The static lock in `ObjcAppLaunchHandler` is enough to secure concurrent access, I was cautious to keep previous behaviour but we might not need it!

Reply:
Sounds good 👍. AFAIK we only observed the TSAN issue happening on CI (when running tests through CLI).